src/content_script/input_tools.js

```diff
@@ -226,15 +24,240 @@ if (!window.hasSLButton) {
+            if (i === -1) {
+              btnWrapper.parentNode.removeChild(btnWrapper);
+              clearInterval(intervalId);
+            }
+
+            // get dimension & position of input
+            const inputCoords = inputElem.getBoundingClientRect();
+            const inputStyle = getComputedStyle(inputElem);
+            const elemWidth = InputTools.dimensionToInt(btnWrapper.style.width);
+            const pageXOffset = window.pageXOffset;
+            const pageYOffset = window.pageYOffset;
+            const buttonXOffset =
+              SLSettings.SLButtonPosition === "right-inside"
+                ? -elemWidth * 1.02
+                : elemWidth * 0.02;
+
+            // calculate elem position
+            const left =
+              InputTools.sumPixel([
+                inputCoords.left,
+                pageXOffset,
+                inputElem.offsetWidth,
+                buttonXOffset,
+                -inputStyle.paddingRight,
+              ]) + "px";
+
+            const top = InputTools.sumPixel([inputCoords.top, pageYOffset]) + "px";
+
+            if (btnWrapper.style.left !== left) {
+              btnWrapper.style.left = left;
+            }
+
+            if (btnWrapper.style.top !== top) {
+              btnWrapper.style.top = top;
+            }
+          }
+
+          intervalId = setInterval(updatePosition, 200);
+          updatePosition();
+        },
 
-  async function initModule() {
-    SLSettings = await sendMessageToBackground("GET_APP_SETTINGS");
-    if (SLSettings.showSLButton) {
-      InputTools.init(document);
-      addMutationObserver();
+        async handleOnClickSLButton(inputElem, slButton) {
+          if (InputTools.isLoading) {
+            return;
+          }
+          InputTools.isLoading = true;
+          slButton.classList.add("loading");
+
+          let res = await sendMessageToBackground("NEW_RANDOM_ALIAS", {
+            currentUrl: window.location.href,
+          });
+          if (res.error) {
+            alert("SimpleLogin Error: " + res.error);
+            res = { alias: "" };
+          }
+
+          InputTools.isLoading = false;
+          slButton.classList.remove("loading");
+
+          inputElem.value = res.alias;
+        },
+
+        sumPixel(dimensions) {
+          let sum = 0;
+          for (const dim of dimensions) {
+            sum += !isNaN(dim) ? dim : InputTools.dimensionToInt(dim);
+          }
+          return sum;
+        },
+
+        dimensionToInt(dim) {
+          try {
+            const pixel = parseFloat(dim.replace(/[^0-9.]+/g, ""));
+            return isNaN(pixel) ? 0 : pixel;
+          } catch (e) {
+            return 0;
+          }
+        },
+      };
+
+      const MutationObserver =
+        window.MutationObserver ||
+        window.WebKitMutationObserver ||
+        window.MozMutationObserver;
+
+      /**
+       * Add DOM mutations listener
+       */
+      const addMutationObserver = () => {
```

Comment:
This script uses `async` functions which [are stable since 2017](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function).

`const` assignments [are stable since Chrome 21](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/const) , which was released [on July 2012](https://en.wikipedia.org/wiki/Google_Chrome_version_history).

So if we're already using `async` functions, using `const` assignments should be a safe option.

Reply:
Oops, this delay was present in the previous version of the extension: https://github.com/simple-login/browser-extension/blob/2a8c23d7369c5699857ff397b22fdcc50834b471/src/background/onboarding.js#L24

Removed it.

Reply:
ok I see, we can use these JS features then.